kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java

```diff
@@ -2,72 +2,93 @@
+  protected KafkaStreams app;
 
-  protected final KafkaStreams app;
+  public KafkaStreamsApp(ConfigClient configClient) {
+    super(configClient);
+  }
 
-  protected KafkaStreamsApp(Config jobConfig) {
-    Properties streamsConfig = getStreamsConfig(jobConfig);
-    getLogger().info(ConfigUtils.propertiesAsList(streamsConfig));
+  @Override
+  protected void doInit() {
+    try {
+      Properties streamsConfig = getStreamsConfig(getAppConfig());
+      getLogger().info(ConfigUtils.propertiesAsList(streamsConfig));
 
-    StreamsBuilder streamsBuilder = new StreamsBuilder();
-    streamsBuilder = buildTopology(streamsConfig, streamsBuilder);
-    Topology topology = streamsBuilder.build();
-    getLogger().info(topology.describe().toString());
+      Map<String, KStream<?, ?>> sourceStreams = new HashMap<>();
+      StreamsBuilder streamsBuilder = new StreamsBuilder();
+      streamsBuilder = buildTopology(streamsConfig, streamsBuilder, sourceStreams);
+      Topology topology = streamsBuilder.build();
+      getLogger().info(topology.describe().toString());
 
-    app = new KafkaStreams(topology, streamsConfig);
+      app = new KafkaStreams(topology, streamsConfig);
 
-    // useful for resetting local state - during testing or any other scenarios where
-    // state (rocksdb) needs to be reset
-    if (streamsConfig.containsKey(CLEANUP_LOCAL_STATE)) {
-      boolean cleanup = Boolean.parseBoolean((String) streamsConfig.get(CLEANUP_LOCAL_STATE));
-      if (cleanup) {
-        getLogger().info("=== Resetting local state ===");
-        app.cleanUp();
+      // useful for resetting local state - during testing or any other scenarios where
+      // state (rocksdb) needs to be reset
+      if (streamsConfig.containsKey(CLEANUP_LOCAL_STATE)) {
+        boolean cleanup = Boolean.parseBoolean((String) streamsConfig.get(CLEANUP_LOCAL_STATE));
+        if (cleanup) {
+          getLogger().info("=== Resetting local state ===");
+          app.cleanUp();
+        }
       }
-    }
 
-    app.setStateListener(new LoggingStateListener(app));
-    app.setGlobalStateRestoreListener(new LoggingStateRestoreListener());
-    app.setUncaughtExceptionHandler((thread, exception) -> {
-          getLogger().error("Thread=[{}] encountered=[{}]. Will exit.", thread.getName(),
-              ExceptionUtils.getStackTrace(exception));
-          System.exit(1);
-        }
-    );
+      app.setStateListener(new LoggingStateListener(app));
+      app.setGlobalStateRestoreListener(new LoggingStateRestoreListener());
+      app.setUncaughtExceptionHandler((thread, exception) -> {
+            getLogger().error("Thread=[{}] encountered=[{}]. Will exit.", thread.getName(),
+                ExceptionUtils.getStackTrace(exception));
+            System.exit(1);
+          }
+      );
+    } catch (Exception e) {
+      getLogger().error("Error initializing - ", e);
+      throw new RuntimeException(e);
```

Comment:
Lets exit from here to after printing stack trace. I see our platform launcher is not exiting with code `1` on error.

Reply:
fixed